core/src/Dscp/Core/Foundation/Witness.hs

```diff
@@ -236,26 +242,50 @@ newtype GTxId = GTxId (Hash GTx)
 
 -- | Compute tx id.
 toGTxId :: (Serialise Tx, Serialise PublicationTx) => GTx -> GTxId
-toGTxId (GMoneyTx tx) = GTxId . unsafeCastHash . toTxId $ tx
+toGTxId (GMoneyTx tx)       = GTxId . unsafeCastHash . toTxId $ tx
 toGTxId (GPublicationTx tx) = GTxId . unsafeCastHash . toPtxId $ tx
 
-data GTxWitnessed
+data WithSize a = WithSize
+    { wsSize    :: Word32
+    , wsPayload :: a
+    }
+    deriving (Generic, Eq, Show)
+
+withSize :: Serialise a => a -> WithSize a
+withSize payload =
+    WithSize (fromIntegral $ BS.length (serialise payload)) payload
+
+instance Buildable a => Buildable (WithSize a) where
+    build (WithSize size a) = "WithSize {" +| size |+ ", " +| a |+ "}"
+
+type GTxWitnessed = WithSize GTxWitnessedRaw
```

Comment:
This doesn't look right.

Reply:
This is not serialisation wrapper, it it from our domain of modelling. We have fees, and we get them from tx size. We can't put size inside the transaction, because tx should know nothing about its size and fees. But the transaction should carry size anyway.

You are free to show better way to do this.

Reply:
Transaction carries size by itself, implicitly.